psi4/driver/json_wrapper.py

```diff
@@ -32,27 +32,206 @@
 import os
 import copy
 import uuid
+import json
 
 import psi4
 from psi4.driver import driver
 from psi4.driver import molutil
+from psi4.driver import p4util
 from psi4 import core
 
-
-methods_dict = {
+methods_dict_ = {
     'energy': driver.energy,
     'gradient': driver.gradient,
-    'property': driver.properties,
+    'properties': driver.properties,
     'optimize': driver.optimize,
     'hessian': driver.hessian,
     'frequency': driver.frequency,
 }
+can_do_properties_ = {
+    "dipole", "quadrupole", "mulliken_charges", "lowdin_charges", "wiberg_lowdin_indices", "mayer_indices"
+}
+
+def _clean_psi_environ(do_clean):
+    if do_clean:
+        psi4.core.clean_variables()
+        psi4.core.clean_options()
+        core.clean()
+
+def run_json(json_data, clean=True):
+
+    # Set scratch
+    if "scratch_location" in json_data:
+        psi4_io = core.IOManager.shared_object()
+        psi4_io.set_default_path(json_data["scratch_location"])
+
+    # Set memory
+    if "memory" in json_data:
+        psi4.set_memory(json_data["memory"])
 
+    # Do we return the output?
+    return_output = json_data.pop("return_output", False)
+    if return_output:
+        outfile = os.path.join(core.IOManager.shared_object().get_default_path(), str(uuid.uuid4()) + ".json_out")
+        core.set_output_file(outfile, False)
+        json_data["raw_output"] = "Not yet run."
+
+    # Set a few flags
+    json_data["raw_output"] = None
+    json_data["success"] = False
+
+    # Attempt to run the computer
+    try:
+        if ("schema_name" in json_data) and (json_data["schema_name"] == "QC_JSON"):
+            run_json_qc_schema(json_data, clean)
+        else:
+            run_json_original_v1_1(json_data, clean)
 
-def run_json(json_data):
+    except Exception as error:
+        json_data["error"] = repr(error)
+        json_data["success"] = False
+
+    if return_output:
+        with open(outfile, 'r') as f:
+            json_data["raw_output"] = f.read()
+        os.unlink(outfile)
+
+    return json_data
+
+
+def run_json_qc_schema(json_data, clean):
+    """
+    An implementation of the QC JSON Schema (molssi-qc-schema.readthedocs.io/en/latest/index.html#) implementation in Psi4.
+
+
+    Parameters
+    ----------
+    json_data : JSON
+        Please see molssi-qc-schema.readthedocs.io/en/latest/spec_components.html for further details.
+
+    Notes
+    -----
+    !Warning! This function is experimental and likely to change in the future.
+    Please report any suggestions or uses of this function on github.com/MolSSI/QC_JSON_Schema.
+
+    Examples
+    --------
+
+    """
+
+    # Clean a few things
+    _clean_psi_environ(clean)
+
+    # This is currently a forced override
+    json_data["schema_version"] = "0"
```

Comment:
@dgasmith, did you have `v0` or `0` in mind for schema_version?

Reply:
`0` as an int I think. If we are going to make it easy to compare versions we should make it really easy, some ideas from [here](https://en.wikipedia.org/wiki/Magic_number_(programming)).

Reply:
ok, just fixed it up here. RTG now, imo.